src/rgw/rgw_op.cc

```diff
@@ -3514,3 +3531,335 @@ void RGWHandler::put_op(RGWOp *op)
   delete op;
 }
 
+
+
+/* Object Rename operation */
+
+int RGWRenameObj::verify_permission()
+{
+    // This function used to check ACLs in legacy code
```

Comment:
Shouldn't set_atomic be done in this verify_permissions only(as done in GetObj::verify_permission), this will remove the need of RGW_ATTR_RENAME_MUTEX (additional metadata on object)